feat: enable DASHBOARD_RBAC

Ian2012 opened the pull request:
## Description

This PR enables DASHBOARD_RBAC and modifies the default role to be compliant with our implementation.

## Steps to test:

Assuming you have more than one course at least 1 event from those courses.

1. Register a new user
2. Grant course staff permissions to the user for just one course.
3. Open superset with this user.
4. Create a dashboard and a chart that queries the `xapi_events_all_parsed` dataset.
5. Assign the role `Open edX` to the chart in the properties.
6. Compare the records the created user has vs the superuser.
7. See the difference.

In the right is the staff user, in the left is the superuser.

![image](https://github.com/openedx/tutor-contrib-superset/assets/33465240/c47bb5c5-e559-4a43-8729-d96f400726b0)

![image](https://github.com/openedx/tutor-contrib-superset/assets/33465240/6b5bf8fe-bad6-4278-9540-917f0e9c26ca)


This confirms that the current implementation of permissions is compatible with the RBAC feature.

This PR fixes:
- https://github.com/openedx/tutor-contrib-oars/issues/27
- https://github.com/openedx/tutor-contrib-oars/issues/15

openedx-webhooks commented:
Thanks for the pull request, @Ian2012!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

<!-- comment:end_of_wip -->




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

Ian2012 commented:
@bmtcril @pomegranited What do you think?

We are still missing the automatic association of roles for the dashboards, so should we contribute to the superset-api-client first?

bmtcril commented:
This seems like a good first step, but I admit I've been much further out of the auth area than @pomegranited . There are two sets of conversations happening right now that might impact this work. The conversations around combining Cairn and OARS, and conversations happening with 2U about edx-rbac which we might use in a future version to have more fine grained roles that aren't possible right now (non-course-staff analytics users, org level analytics users, etc).

I'm hoping we can have a conversation with Regis about what he's doing on the Cairn side in the next couple of days to make sure we're aligned on direction and implementation.

Ian2012 commented:
@pomegranited I just "migrated" the Gamma permissions to the Open edX role and removed the auto assign gamma role.

I also removed access to the Data, Dataset, and Databases menus, as those shouldn't be viewed by those users. I think we can keep the gamma user role for Data Analyst and future supported roles, and Superset do not recommend modifying the default roles https://superset.apache.org/docs/security/#provided-roles

As there is not really a specification for what an Open edX user can do, what do you think we need to **restrict/allow** by default?

I was thinking of removing the SQL Query, Chart and Dashboard creation access but I think is important that users can create their own dashboards (with only the information they have access of course).

![image](https://github.com/openedx/tutor-contrib-superset/assets/33465240/5b1b25a4-9ed9-470f-b3d4-b5b0bb8d59e3)

mariajgrimaldi commented on tutorsuperset/templates/superset/apps/data/roles.json:
```diff
@@ -3989,6 +3989,1142 @@
         "view_menu":{
           "name":"Superset"
         }
+      },
```
I'm not familiar with this file. What does it do?

Ian2012 replied:
It's a file with the permission definition for every role. I just copied the Gamma role to grant minimal permissions and removed what I considered was not right for the default course staff user

mariajgrimaldi commented:
So `dataset level checks` doesn't refer to row level security? That's interesting and confusing 😅

pomegranited reviewed (approved):
@Ian2012 This is working as expected, thank you!

Couple of nits, and a a minor note:

* Enabling this setting requires that we set a role for each dashboard.. Non-admin users see an weird error like this when viewing a dashboard they don't have access to:

  ![rbac_on_no_role](https://github.com/openedx/tutor-contrib-superset/assets/7556571/2ecd6a75-74aa-4c6d-a30e-50f8dcb8f057)

:+1:

- [x] I tested this as described in the description, and it worked a treat. I did have to manually remove the Gamma role from existing Superset user records, but no big deal.
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A

Ian2012 commented:
@bmtcril I added documentation and this is ready for review

openedx-webhooks commented:
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-superset&entry.867055334=https://github.com/openedx/tutor-contrib-superset/pull/42&entry.1484655318=https://github.com/Ian2012&entry.752974735=2023-05-25+22:06&entry.1917517419=2023-06-02+13:31&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
